src/posts/2021-11-19-ecto_query_cheatsheet.md

```diff
@@ -0,0 +1,180 @@
+---
+title: My Ecto.Query Cheatsheet
+description: A collection of things I try not to google twice when working on Dispatch
+author: jennablumenthal
+date: 2021-11-19
+tags:
+  - posts
+layout: layouts/post.njk
+---
+
+#### Preamble
+
+This is a collection of the stuff I was looking for when I first began dabbling in Elixir, Phoenix, and particularly Ecto (the <a href="https://hexdocs.pm/ecto/Ecto.html" target="_blank">library</a> used in Phoenix to interact with the data store). It is by no means a comprehensive tutorial, but rather a set of common CRUD-type operations I kept googling over and over until I finally wrote them down somewhere.
+
+If you'd like, you can clone and run the <a href="https://github.com/bikebrigade/dispatch" target="_blank">Dispatch</a> repo to code along with the examples.
+
+##### A note on syntax
+
+Ecto allows you to write queries in two ways: **keyword-based** and **macro-based**. Most examples in the [official docs](https://hexdocs.pm/ecto/Ecto.Query.html) use keyword-based examples, but I find the less verbose macro-based more straightforward for simple operations.
+
+It's hard to argue against the value of writing queries in an almost-identical-to-SQL format, I think this was part of the reason I found Ecto a bit "fiddly" to get started with.
```

Comment:
I think these two clauses contradict each other: the almost-identical-to-SQL format is a positive, finding it "fiddly" is a negative do you mean that the fiddlyness came from it not being exactly SQL, or from not being used to SQL?

Reply:
Hmm yeah. I found the fiddly-ness that you have to write a whole SQL statement for something simple like "find first". I'll try to reword/explain that better. Ideas welcome!